### PR TITLE
Saas 17.4 mail message format scju

### PR DESCRIPTION
**Pull request by jaeschwa**
[IMP] *: Improvment of the mail messages

In all the modules improved the message related to the on boarding.
Modified mail messages, fixed redirection, the automatic opening and
the purple dot are not implemented.

Changed the tests accordingly with the new mail message definition.

task-4056239

### CODE REVIEW

File: tests/test_generic/tests/test_industry_requirements.py [hunk 1]
```diff
@@ -56,11 +56,6 @@ def test_knowledge_article_notification(self):
                 0,
                 "The knowledge article should be in the favorite category",
             )
-            self.assertIn(
-                'href="/knowledge/article/%s' % knowledge_article.id,
-                notif.body,
-                "The notification link should target the module-related knowledge article.",
-            )
```
Comment (vava-odoo) [inline tests/test_generic/tests/test_industry_requirements.py:63]: keep it please

File: tests/test_generic/tests/test_industry_requirements.py [hunk 2]
```diff
@@ -70,6 +65,8 @@ def test_cloc_exclude_view(self):
         c.report(True)  # show details of cloc in the logs
         for module in self.installed_modules:
             for cloc_entry in c.modules.get(module, {}):
+                if 'src/js/' in cloc_entry:
+                    continue
```
Comment (vava-odoo) [inline tests/test_generic/tests/test_industry_requirements.py:69]: No. We need it. It should work by adding a cloc_exclude for this ir.asset

File: tests/test_generic/tests/test_manifest.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
     'license': '',
     'name': '',
     'version': '',
+    'assets':{},
```
Comment (vava-odoo) [inline tests/test_generic/tests/test_manifest.py:24]: :innocent: 
```suggestion
    'assets': {},
```

File: tests/test_generic/tests/test_xml.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ def _check_files_in_path(self, module):
             for file_name in files:
                 file_path = os.path.join(root, file_name)
                 ext = os.path.splitext(file_path)[1].lower()
-                if 'static/' in file_path and 'description' not in file_path:
+                if not file_path.endswith('.js') and ('static/' in file_path and 'description' not in file_path):
```
Comment (vava-odoo) [inline tests/test_generic/tests/test_xml.py:37]: Couldn't we also check that the tour exists?

File: agriculture_shop/data/mail_message.xml [hunk 5]
```diff
@@ -6,11 +6,8 @@
...
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Agriculture Shop</field>
         <field name="body" model="knowledge.article" eval="
-            '&lt;br>Hey! Need help?&lt;/br>
-             &lt;br>Checkout this &lt;a href=\''
-             + obj().env.ref('agriculture_shop.knowledge_article_1').article_url.replace(obj().get_base_url(), '')
-             + '\'>article&lt;/a>&lt;/br>'"/>
+        '&lt;span>&#x1F44B; Hi! Follow this &lt;a href=\'/odoo/knowledge/33\'>onboarding guide&lt;/a>. You can find it anytime in the Knowledge app.&lt;/span>'"/>
```
Comment (vava-odoo) [inline agriculture_shop/data/mail_message.xml:11]: 33 is valid for you, but surely not for all dbs. You need to use something like `obj().env.ref('agriculture_shop.knowledge_article_1')`

File: agriculture_shop/static/src/js/my_tour.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  rainbowManMessage: _t("Congrats :)"),
+  sequence: 2,
+  steps: () => [
+    {
+      trigger: '.o_app[data-menu-xmlid="knowledge.knowledge_menu_root"]',
+      content: markup(
+        _t(
+          "Open Knowledge to find the onboarding guide."
+        )
+      ),
```
Comment (vava-odoo) [inline agriculture_shop/static/src/js/my_tour.js:18]:
```suggestion
      content: markup(_t("Open Knowledge to find the onboarding guide.")),
```

File: agriculture_shop/static/src/js/my_tour.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { markup } from "@odoo/owl";
+
+registry.category("web_tour.tours").add("knowledge_tour", {
+  url: "/web",
```
Comment (vava-odoo) [inline agriculture_shop/static/src/js/my_tour.js:8]: I think it changed before 17.4, so /web will be deprecated soon
```suggestion
  url: "/odoo",
```

File: agriculture_shop/static/src/js/my_tour.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  rainbowManMessage: _t("Congrats :)"),
```
Comment (vava-odoo) [inline agriculture_shop/static/src/js/my_tour.js:9]: Either you add a step to show it, either you remove this line (I'd go for option 2)
```suggestion
```

**Review by vava-odoo — CHANGES_REQUESTED** [review]
Looks overall good, but noted a few things to improve. Thanks @jaeschwa

File: tests/test_generic/tests/test_xml.py (hunk 4 above)
Comment (jaeschwa) [reply tests/test_generic/tests/test_xml.py:37]: What do you mean the tour exists?

File: agriculture_shop/static/src/js/my_tour.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { markup } from "@odoo/owl";
+
+registry.category("web_tour.tours").add("knowledge_tour", {
+  url: "/odoo",
+
```
Comment (vava-odoo) [inline agriculture_shop/static/src/js/my_tour.js:9]:
```suggestion
```

File: construction/data/knowledge_article.xml [hunk 10]
```diff
@@ -1,6 +1,6 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <odoo>
-        <record id="knowledge_article_35" model="knowledge.article">
+        <record id="welcome_article5" model="knowledge.article">
```
Comment (vava-odoo) [inline construction/data/knowledge_article.xml:3]:
```suggestion
        <record id="welcome_article" model="knowledge.article">
```

File: micro_brewery/data/ir_model_data.xml [hunk 11]
```diff
@@ -6,10 +6,4 @@
          <field name="module">__cloc_exclude__</field>
          <field name="res_id" eval="ref('contactus')"/>
      </record>
-     <record id="cloc_exclude_appointment_heading" model="ir.model.data">
-        <field name="name">cloc_exclude_appointment_heading</field>
-        <field name="model">ir.ui.view</field>
-        <field name="module">__cloc_exclude__</field>
-        <field name="res_id" eval="ref('micro_brewery_custom_appointment_heading')"/>
-    </record>
```
Comment (vava-odoo) [inline micro_brewery/data/ir_model_data.xml:14]: Doesn't seem related to this task

File: tests/test_generic/tests/test_xml.py (hunk 4 above)
Comment (vava-odoo) [reply tests/test_generic/tests/test_xml.py:37]: I'd add a few tests to make sure we don't forget all these must-have
- id welcome_article
- js tour in test_generic
- check the mail message 
- ...

File: agriculture_shop/static/src/js/my_tour.js (hunk 9 above)
Comment (vava-odoo) [reply agriculture_shop/static/src/js/my_tour.js]: Please apply everywhere...

**Comment by vava-odoo** [comment]
See https://github.com/odoo/industry/pull/256
